profiling-ffi/src/profiles.rs

```diff
@@ -320,11 +320,9 @@ pub unsafe extern "C" fn ddog_prof_Profile_new(
 /// The `profile` can be null, but if non-null it must point to a valid object
 /// created by the Rust Global allocator.
 #[no_mangle]
-pub unsafe extern "C" fn ddog_prof_Profile_drop(profile: Option<&mut Profile>) {
-    if let Some(reference) = profile {
-        // Safety: Profile is not repr(C), and is therefore boxed.
-        drop(Box::from_raw(reference as *mut _))
-    }
+pub unsafe extern "C" fn ddog_prof_Profile_drop(profile: *mut Profile) {
+    // Safety: Profile is not repr(C), and is therefore boxed.
+    drop(Box::from_raw(profile))
```

Comment:
Help me understand this one. I don't get why this is better, partly because it will not handle a nullptr.

Reply:
Good catch on the `nullptr`. I had actually done this in the playground when I was testing out the code, but forgot to copy that back to the main fix.

The issue is that its not legal to `drop` a `&mut`: to quote *Rust for Rustacians* p11: "the difference between owning a value and having a mutable reference is that the owner is responsible for dropping the value when it is no longer necessary ... if you move the value behind a mutable reference, you must leave another value in its place".  Miri identified that several tests were doing UB because of this: changing to a pointer made miri happy.  Playground link: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=b3fb4e106c1d19183c831f38389f2658

If you have a better proposal, I'm happy to do that instead.  But I'd rather not leave UB hanging around.